### ISSUE TITLE
bottles#list doesn't return tiny view

### ISSUE THREAD

**Issue by barracks510** [raphael/goa-cellar#15]
In the design package, I believe the list action should return a collection of tiny bottles, but in the controller code and in the generated swagger api, the list action returns a collections of bottles in the default view. If this is the intended behaviour, how exactly should one specify that a resource returns a collection of tiny views?

**Comment by lsh246** [comment]
Yeah, I had this question also and asked it on the goa Slack channel. To summarize my question, the DSL for the `List` method on the `Bottle` resource includes this for the response:

```
		Response(OK, func() {
			Media(CollectionOf(Bottle, func() {
				View("default")
				View("tiny")
			}))
		})
```

My understanding of what this means is: It's not that the `List` action should **only** return a collection of tiny-view bottles. Rather, it can return **either** a collection of default-view **or** a collection of tiny-view bottles. The current implementation in the bottle controller (in `bottle.go`) only returns a collection of default-view bottles, but theoretically it could be enhanced to return one or the other depending on the request (via say a query string parameter).

Raphael's original answer is: 

>the idea is that there is one response method generated per view. Then the controller code decides which one to use depending on the request.
>in goa-cellar the list account action uses the tiny view to render the elements of the response array while the show action uses the default view. That's just an example you could also define a query string parameter to decide on which view to use.

**Comment by barracks510** [comment]
I see. If there were circular definitions, thus requiring that only a "tiny" view being shown, do you know what the proper way of writing _that_ DSL might look like?
